### PR TITLE
dataflow-types: Add protobuf support for Peek

### PR DESCRIPTION
**Pull request by lluki**
Add Protobuf support for `Peek` client command.

Related to #11971

For full fix, requires implementing #11970 

### Tips for reviewer

Possibly the `any_uuid` strategy could be moved to a better location.

### Testing

 - Testing is implemented, but tests are currently `#[ignore]` because they need #11970 (SafeMfpPlan) to be correct.

### Release notes

None

### CODE REVIEW

File: src/dataflow-types/src/client.rs [hunk 1]
```diff
@@ -96,6 +102,52 @@ pub struct Peek<T = mz_repr::Timestamp> {
     pub map_filter_project: mz_expr::SafeMfpPlan,
 }
 
+impl<T: PartialEq> PartialEq for Peek<T> {
```
Comment (wangandi) [inline src/dataflow-types/src/client.rs:105]: Could PartialEq not be automatically derived for Peek<T>?

File: src/repr/src/row.rs [hunk 2]
```diff
@@ -99,6 +101,42 @@ pub struct Row {
...
+    type Strategy = BoxedStrategy<Row>;
+
+    fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
+        let datums = prop_oneof![
+            Just(Datum::Null),
+            Just(Datum::False),
+            Just(Datum::Int16(0)),
+            Just(Datum::Int32(0)),
+            Just(Datum::Int64(0)),
+            Just(Datum::Float32(OrderedFloat(0.0))),
```
Comment (wangandi) [inline src/repr/src/row.rs:115]: I think we should add a TODO to allow more than one value for each Datum type. (Other than Datum::Null, Datum::False, etc.)

File: src/repr/src/row.rs [hunk 3]
```diff
@@ -99,6 +101,42 @@ pub struct Row {
     data: SmallVec<[u8; Self::SIZE]>,
 }
 
+impl Arbitrary for Row {
+    type Parameters = ();
+    type Strategy = BoxedStrategy<Row>;
+
+    fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
```
Comment (wangandi) [inline src/repr/src/row.rs:108]: What if we had `type Parameters = usize` so we could specify the number of datums we want in the row? So far this only allows creating a row with a single number.

File: src/repr/src/proto.rs [hunk 4]
```diff
@@ -155,6 +156,21 @@ impl ProtoRepr for u128 {
     }
 }
 
+impl ProtoRepr for Uuid {
+    type Repr = ProtoUuid;
```
Comment (wangandi) [inline src/repr/src/proto.rs:160]: Why not translate `Uuid` to `ProtoU128`?

File: src/repr/src/row.rs (hunk 2 above)
Comment (lluki) [reply src/repr/src/row.rs:115]: I assume this comment will be adressed in #12125

File: src/repr/src/row.rs (hunk 3 above)
Comment (lluki) [reply src/repr/src/row.rs:108]: I assume this comment will be adressed in #12125

File: src/dataflow-types/src/client.rs (hunk 1 above)
Comment (lluki) [reply src/dataflow-types/src/client.rs:105]: You are right, this works.

**Pull request merged**